pkg/goDB/DBWorkManager.go

```diff
@@ -499,14 +603,31 @@ func (w *DBWorkManager) grabAndProcessWorkload(ctx context.Context, wg *sync.Wai
 			}
 		}()
 
+		var lastStatsUpdate time.Time
 		for workload := range workloadChan {
+			var stats = new(WorkloadStats)
 			resultMap := hashmap.NewAggFlowMapWithMetadata()
 			for _, workDir := range workload.workDirs {
+				// printing of status information
+				if w.keepAlive > 0 {
+					workload.statsCallbacks = StatsCallbacks{
+						func(s *WorkloadStats) error {
+							logWorkloadStats(
```

Comment:
Why not have `logWorkloadStats` return an error (always `nil`), then you can save yourself the unneeded wrapper `func()` here...?

Reply:
The wrapper is still needed to inject the logger.

Reply:
Nevermind. Completely simplified it.